tests/core/pyspec/eth2spec/test/electra/block_processing/test_process_consolidation_request.py

```diff
@@ -840,11 +1138,75 @@ def run_consolidation_processing(spec, state, consolidation, success=True):
         assert state.validators[source_index].exit_epoch < spec.FAR_FUTURE_EPOCH
         # Check that the exit epoch matches earliest_consolidation_epoch
         assert state.validators[source_index].exit_epoch == state.earliest_consolidation_epoch
+        # Check that the withdrawable_epoch is set correctly
+        assert state.validators[source_index].withdrawable_epoch == (
+            state.validators[source_index].exit_epoch + spec.config.MIN_VALIDATOR_WITHDRAWABILITY_DELAY
+        )
         # Check that the correct consolidation has been appended
         expected_new_pending_consolidation = spec.PendingConsolidation(
             source_index=source_index,
             target_index=target_index,
         )
         assert state.pending_consolidations == pre_pending_consolidations + [expected_new_pending_consolidation]
+        # Check excess balance is queued if the target switched to compounding
+        if pre_target_withdrawal_credentials[:1] == spec.ETH1_ADDRESS_WITHDRAWAL_PREFIX:
+            assert state.validators[target_index].withdrawal_credentials == (
+                spec.COMPOUNDING_WITHDRAWAL_PREFIX + pre_target_withdrawal_credentials[1:])
+            assert state.balances[target_index] == spec.MIN_ACTIVATION_BALANCE
+            if pre_target_balance > spec.MIN_ACTIVATION_BALANCE:
+                assert state.pending_balance_deposits == [spec.PendingBalanceDeposit(
+                    index=target_index, amount=(pre_target_balance - spec.MIN_ACTIVATION_BALANCE))]
+        else:
+            assert state.balances[target_index] == pre_target_balance
+    else:
+        assert pre_state == state
+
+
+def run_switch_to_compounding_processing(spec, state, consolidation, success=True):
+    """
+    Run ``process_consolidation``, yielding:
+      - pre-state ('pre')
+      - consolidation_request ('consolidation_request')
+      - post-state ('post').
+    If ``success == False``, ``process_consolidation_request`` would return without any state change.
+    """
+
+    if success:
+        validator_pubkeys = [v.pubkey for v in state.validators]
+        source_index = spec.ValidatorIndex(validator_pubkeys.index(consolidation.source_pubkey))
+        target_index = spec.ValidatorIndex(validator_pubkeys.index(consolidation.target_pubkey))
+        source_validator = state.validators[source_index]
```

Comment:
Out of caution, we should add two asserts to ensure these validators are present. Might prevent a bad test. Otherwise, the index will be `-1` and `source_validator` will be set to the last validator. Something like:

```python
assert consolidation.source_pubkey in validator_pubkeys
assert consolidation.target_pubkey in validator_pubkeys
```

We could/should do the same thing in `run_consolidation_processing` too.